fvtr/fvtr.sh

```diff
@@ -150,28 +150,15 @@ then
 	exit 1
 fi
 
-# creating a file that will hold the AT_* vars
-# scripts that need to use any of the AT_* vars can
-# access them by calling:
-# array set AT_ENV [ exec cat envvararray.txt ]
-# i.e. $AT_ENV(AT_DEST)
-#
 source $CONFIGFILE
-set > envtempfile.txt
-grep "AT_" envtempfile.txt | sed 's/^> //' | sed "s/'/{/" | \
-	sed "s/'/}/"| sed 's/=/!!!/' > envvars.txt
-while read line
-do
-	varname=""
-	varval=""
-	varname=$(echo $line | awk -F'!!!' '{ print $1 }')
-	varval=$(echo $line |  awk -F'!!!' '{ print $2 }')
-	if [ ${#varval} -eq 0 ]
-	then
-		varval="NOT_SET"
-	fi
-	export $varname="$varval"
-done < envvars.txt
+eval $(
+    set | grep "^AT_" | sed -e "s/^> //" -e "s/'/{/" -e "s/'/}/" -e "s/=/ /" | \
+	while read varname varval
+	do
+	    test -z "$varval" && varval="NOT_SET"
+	    echo "export $varname=\"$varval\""
```

Comment:
Is the `echo` call correct, here?  What effect does it have?

Reply:
@gftg85 
In its subshell, `echo` will send to stdout many strings like the following one:
`export AT_VARNAME="A_VALUE"`
The `eval` will evaluate those strings. The variable `AT_VARNAME` will be exported in the current shell (the script one) environment.

If the `export` is done in the while-loop, the variable will be exported in the subshell environment only.

Reply:
Ha!  So that's how it works.  Thanks for the explanation.